pkg/operator/controller/externaldns/deployment.go

```diff
@@ -363,19 +363,14 @@ func externalDNSAnnotationsChanged(current, expected, updated *appsv1.Deployment
 func externalDNSContainersChanged(current, expected, updated *appsv1.Deployment) bool {
 	changed := false
 
-	// number of container is different: let's reset them all
-	if len(current.Spec.Template.Spec.Containers) != len(expected.Spec.Template.Spec.Containers) {
-		updated.Spec.Template.Spec.Containers = expected.Spec.Template.Spec.Containers
-		return true
-	}
-
 	currentContMap := buildIndexedContainerMap(current.Spec.Template.Spec.Containers)
 	expectedContMap := buildIndexedContainerMap(expected.Spec.Template.Spec.Containers)
```

Comment:
Is it ok to not have a container map for the `updated.Spec.Template.Spec.Containers`?  You are mixing assignment by index with append in L397, within the same loop - are you sure that's going to work for every scenario?

Reply:
> Is it ok to not have a container map for the `updated.Spec.Template.Spec.Containers`?

Yes, the current and the updated are the same in terms of the containers (what this function is supposed to do) at the beginning of the function. That's actually the whole point: `updated == current + updated things`.

> You are mixing assignment by index with append in L397, within the same loop - are you sure that's going to work for every scenario?

That's an interesting finding, didn't think of it indeed. Good enough it doesn't impact the algorithm: the indexes are still pointing to the right containers as we append to the end of the container list and the index container map is not refreshed.

Reply:
Okay